spec/datadog/appsec/waf/context_spec.rb

```diff
@@ -202,96 +177,43 @@
         end
       end
     end
-  end
 
-  context 'with a partially bad ruleset' do
-    let(:rule) do
-      {
-        'version' => '2.2',
-        'metadata' => {
-          'rules_version' => '1.2.3'
-        },
-        'rules' => [
-          {
-            'id' => 1,
-            'name' => 'Rule 1',
-            'tags' => { 'type' => 'flow1' },
-            'conditions' => [
-              {
-                'operator' => 'match_regex',
-                'parameters' => { 'inputs' => [{ 'address' => 'value1' }], 'regex' => 'badregex(' }
-              },
-            ],
-            'action' => 'record',
-          },
-          {
-            'id' => 2,
-            'name' => 'Rule 2',
-            'tags' => { 'type' => 'flow2' },
-            'conditions' => [
-              {
-                'operator' => 'match_regex',
-                'parameters' => { 'inputs' => [{ 'address' => 'value2' }], 'regex' => 'rule2' }
-              },
-            ],
-            'action' => 'record',
-          }
-        ]
-      }
-    end
+    context 'stress testing' do
+      let(:run_count) { 500 }
+      let(:thread_count) { 200 }
```

Comment:
One thing: I would avoid putting stress tests in our normal test suite - they are resource intensive and usually take some time to run.

I would propose to extract them to a separate file, and add a tag for them, so we could run them separately from the main test suite:
https://rspec.info/documentation/3.2/rspec-core/file.Filtering.html

Reply:
As a second option, we could use tags to separate them. Don't think we will have a lot now, but it's a less changes option.

https://rspec.info/features/3-12/rspec-core/command-line/tag/

Reply:
Fixed in c4eff93620f0349df72d2d57ea384336cd58b3e5 !